### DISCUSSION TITLE
Support for time-dependent variables and [start, stop) format

### DISCUSSION THREAD

**Discussion by zhiweill** [mlr-org/mlr3proba#371] [Q&A] [answered]
In survival package, the most common way to encode time-dependent covariates is to use the (start, stop] form
of the model: `Surv(time1, time2, status)`. Below is what the [timedep vignettes](https://cran.r-project.org/web/packages/survival/vignettes/timedep.pdf) of survival package said:
<img width="813" alt="Screenshot 2024-02-13 at 5 38 32 PM" src="https://github.com/mlr-org/mlr3/assets/112448023/39cfd3d0-6c93-4054-8cb0-c779349213b8">

In mlr3, `as_task_surv()` do have parameters `time2`, for example, `as_task_surv(mydata, time = "time1",time2 = "time2", event = "status")`.  However, when I run the codes below, the selected feature included the `time2`. 
```
lrn = lrn("surv.cv_glmnet")
lrn$train(mytask, split$train)
lrn$selected_features()
```
Does it mean the surv learners like glmnet regard the `time2` as the predictor instead of the response variable `Surv(...)`? What survival models in mlr3 support `Surv(time1, time2, status)` encoding?

Thanks!!!

**Comment by bblodfon** [comment]
Hi! We do not support anything other than right-censored survival tasks atm. So yes, I think right now `time2` is used as a variable in `surv.cv_glmnet` via `mlr3`. **Changes are needed in compatible learners to understand this time-dependent format** even though the formulation of the `SurvTask` in `mlr3proba` works atm.

To implement this I need either toooooooo much free time :) or a research student to help! In this [issue](https://github.com/mlr-org/mlr3proba/issues/183) I have more details about how this should look like (different types of survival tasks, etc.)

**Reply by zhiweill** [reply]
Thanks for your reply :)

**Reply by agalecki** [reply]
Hi, It appears it works now. Just add `type = "counting"` argument in your task definition 

```
library(mlr3)
library(mlr3proba)
mytask = as_task_surv(survival::heart, time = "start", time2 = "stop", event = "event", type="counting")
mytask
```
```
<TaskSurv:survival::heart> (172 x 8)
* Target: start, stop, event
* Properties: -
* Features (5):
  - dbl (4): age, id, surgery, year
  - fct (1): transplant
> 
```

**Reply by zhiweill** [reply]
Thank you for the note :)

**Comment by bblodfon** [comment, marked as answer]
Hi,

A note for future reference: `mlr3proba` and most probably its successor, `mlr3survival`, will only understand right-censoring survival data/outcomes as this is the most common censoring type encountered for ML.

BR, John
